core/src/main/java/org/apache/iceberg/rest/ErrorHandlers.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.util.function.Consumer;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.BadRequestException;
+import org.apache.iceberg.exceptions.ForbiddenException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.NotAuthorizedException;
+import org.apache.iceberg.exceptions.RESTException;
+import org.apache.iceberg.exceptions.ServiceFailureException;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+
+/**
+ * A set of consumers to handle errors for requests for table entities or for namespace entities,
+ * to throw the correct exception.
+ */
+public class ErrorHandlers {
+
+  private ErrorHandlers() {
+  }
+
+  public static Consumer<ErrorResponse> namespaceErrorHandler() {
+    return baseNamespaceErrorHandler().andThen(defaultErrorHandler());
+  }
+
+  public static Consumer<ErrorResponse> tableErrorHandler() {
+    return baseTableErrorHandler().andThen(defaultErrorHandler());
+  }
+
+  /**
+   * Table level error handlers.
+   * Should be chained wih the {@link #defaultErrorHandler}, which takes care of common cases.
+   */
+  private static Consumer<ErrorResponse> baseTableErrorHandler() {
+    return error -> {
+
+      switch (error.code()) {
+        // HttpStatus.SC_NOT_FOUND
+        case 404:
+          if (NoSuchNamespaceException.class.getSimpleName().equals(error.type())) {
+            throw new NoSuchNamespaceException("Resource not found: %s", error);
+          } else {
+            throw new NoSuchTableException("Resource not found: %s", error);
+          }
+        // HttpStatus.SC_CONFLICT
+        case 409:
+          throw new AlreadyExistsException("The table already exists: %s", error);
```

Comment:
Nit: "Table already exists" rather than "The table ..."

Reply:
Oh woops I already updated this to just be `error.message()`. I'll revert it.